src/elements/content-picker/ContentPicker.js

```diff
@@ -795,7 +797,7 @@ class ContentPicker extends Component<Props, State> {
 
             // Check if we hit the selection limit
             // Ignore when in single file selection mode.
-            if (hasHitSelectionLimit && !isSingleFileSelection) {
+            if (hasHitSelectionLimit && !isSingleFileSelection && !selectedKeys.indexOf(cacheKey)) {
```

Comment:
This new condition seems unrelated to the selection limit. Can we update the comment or break it out into a variable so it's clear what we want to test?

Reply:
I'll do both thanks